arangod/Statistics/StatisticsWorker.cpp

```diff
@@ -1049,9 +1049,8 @@ void StatisticsWorker::run() {
       LOG_TOPIC(WARN, Logger::STATISTICS) << "caught unknown exception in StatisticsWorker";
     }
 
-    seconds++;
+    seconds++; // TODO should this run adaptively ?
+    std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
I looked up both sleep_for and previous condition timed_wait.  Both have implementations that fail to filter signals.  So the timing could be less than full seconds.  In fact, we know that both could be longer (much longer) than a single second.  Should fix this loop to recalculate duration elapsed on each loop.

Reply:
Also, the "seconds" thing above does not take into account how long the individual actions called by `run` (`historian`, `historianAverage` and `collectGarbage`) take. So the name "seconds" here should not be interpreted literally. Maybe we should just rename it to "iterations", and then it would at least be named appropriately.

I don't think it is required for the StatisticsWorker to fire at exactly every elapsed second. The motivation here was more to make the StatisticsWorker yield to other threads so it does not use too much CPU. So it is not required that the StatisticsWorker has a completely accurate timing. But if the accuracy can be improved easily without making it use much more CPU, then I am completely fine with it.